### PR TITLE
corrected sprite.h + finished safari_zone, list_menu, menu_indicators, field_poison, buy_menu_helpers, dynamic_placeholder_text_util

### PR DESCRIPTION
**Pull request by jiangzhengwenjz**

### CODE REVIEW

File: asm/field_control_avatar.s [hunk 1]
```diff
@@ -1901,8 +1901,8 @@ _0806D780: .4byte 0x00004021
 _0806D784: .4byte gPlayerParty
 	thumb_func_end AdjustFriendship_step
 
-	thumb_func_start overworld_poison_timer_set
-overworld_poison_timer_set: @ 806D788
+	thumb_func_start DoPoisonFieldEffect_timer_set
```
Comment (garakmon) [inline asm/field_control_avatar.s]: these functions should be all caps

Comment (garakmon) [reply asm/field_control_avatar.s]: this one specifically looks like `ClearPoisonStepCounter` from pokeemerald and the next is `UpdatePoisonStepCounter`

Comment (jiangzhengwenjz) [reply asm/field_control_avatar.s]: Yes, but I only did field_poison and ported the function names called by functions there. field_control_avatar.s is a large file and is related to `eventObjects` (which should be soon renamed and ported from pokeem by you) so I temporarily tend not to touch it.

Comment (garakmon) [reply asm/field_control_avatar.s]: okay

Comment (jiangzhengwenjz) [reply asm/field_control_avatar.s]: and sure I can correct DoPoisonFieldEffect_timer_set

File: src/list_menu.c [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+        break;
+    case 3:
+        if (list->taskId == TASK_NONE)
+            list->taskId = ListMenuAddCursorObject(list, 1);
+        ListMenuUpdateCursorObject(list->taskId, GetWindowAttribute(list->template.windowId, WINDOW_TILEMAP_LEFT) * 8 + x, GetWindowAttribute(list->template.windowId, WINDOW_TILEMAP_TOP) * 8 + y, 1);
+        break;
+    }
+}
+
+#undef TASK_NONE
```
[the remark this answers is not in the document]
Comment (jiangzhengwenjz) [reply src/list_menu.c]: Do you mean I never need to undef this kind of macro if not necessary?

[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/list_menu.c]: Yes. Macro definitions are not carried between C files.

**Comment by jiangzhengwenjz** [comment]
I put `TASK_NONE` in task.h. Everything is good now

**Pull request merged**